ext/opentelemetry-ext-aiopg/src/opentelemetry/ext/aiopg/__init__.py

```diff
@@ -0,0 +1,132 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+The integration with PostgreSQL supports the aiopg library,
+it can be enabled by using ``AiopgInstrumentor``.
+
+.. aiopg: https://github.com/aio-libs/aiopg
+
+Usage
+-----
+
+.. code-block:: python
+
+    import aiopg
+    from opentelemetry import trace
+    from opentelemetry.ext.aiopg import AiopgInstrumentor
+    from opentelemetry.sdk.trace import TracerProvider
+
+    trace.set_tracer_provider(TracerProvider())
+
+    AiopgInstrumentor().instrument()
+
+    cnx = await aiopg.connect(database='Database')
+    cursor = await cnx.cursor()
+    await cursor.execute("INSERT INTO test (testField) VALUES (123)")
+    cursor.close()
+    cnx.close()
+
+    pool = await aiopg.create_pool(database='Database')
+    cnx = await pool.acquire()
+    cursor = await cnx.cursor()
+    await cursor.execute("INSERT INTO test (testField) VALUES (123)")
+    cursor.close()
+    cnx.close()
+
+API
+---
+"""
+
+import aiopg
+
+from opentelemetry.ext.aiopg import wrappers
+from opentelemetry.ext.aiopg.version import __version__
+from opentelemetry.instrumentation.instrumentor import BaseInstrumentor
+from opentelemetry.trace import get_tracer
+
+
+class AiopgInstrumentor(BaseInstrumentor):
+    _CONNECTION_ATTRIBUTES = {
+        "database": "info.dbname",
+        "port": "info.port",
+        "host": "info.host",
+        "user": "info.user",
+    }
+
+    _DATABASE_COMPONENT = "postgresql"
+    _DATABASE_TYPE = "sql"
+
+    def _instrument(self, **kwargs):
+        """Integrate with PostgreSQL aiopg library.
+           aiopg: https://github.com/aio-libs/aiopg
+        """
+
+        tracer_provider = kwargs.get("tracer_provider")
+
+        tracer = get_tracer(__name__, __version__, tracer_provider)
```

Comment:
IIRC the tracer should be created at span-creation time so if the span is a root span it will have a seperate trace id

Reply:
It was made like in psycopg2 extension, there tracer created in the same place.

Reply:
maybe it shouldn't be like that in psycopg2 either, but I guess thats outside of the scope of this PR

Reply:
In connection with the latest changes in dbapi related to work with traces in this extension I've changed it too. Now tracer is being created in `AsyncTracedCursor.traced_execution` at span-creation time.